src/cmd/flux-mini.py

```diff
@@ -1611,13 +1627,88 @@ def init_jobspec(self, args):
             broker_opts=list_split(args.broker_opts),
             exclusive=args.exclusive,
         )
-        if sys.stdin.isatty():
+
+        #  For --bg, always allocate a pty, but not interactive,
+        #   since an interactive pty causes the job shell to hang around
+        #   until a pty client attaches, which may never happen.
+        #
+        #  O/w, allocate an interactive pty only if stdin is a tty
+        #
+        if args.bg:
+            jobspec.setattr_shell_option("pty", 1)
+        elif sys.stdin.isatty():
             jobspec.setattr_shell_option("pty.interactive", 1)
         return jobspec
 
+    @staticmethod
+    def log(jobid, ts, msg):
+        print(f"{jobid}: {ts:6.3f}s: {msg}", file=sys.stderr, flush=True)
+
+    def bg_wait_cb(self, future, args, jobid):
+        """
+        Wait for memo event, connect to child instance, and finally wait
+        for rc1 to complete
+        """
+        event = future.get_event()
+        if not event:
+            #  The job has unexpectedly exited since we're at the end
+            #   of the eventlog. Run `flux job attach` since this will dump
+            #   any errors or output, then raise an exception.
+            os.system(f"flux job attach {jobid} >&2")
+            raise OSError(f"{jobid}: unexpectedly exited")
+
+        if not self.t0:
+            self.t0 = event.timestamp
+        ts = event.timestamp - self.t0
+
+        if args.verbose and event.name == "alloc":
+            self.log(jobid, ts, "resources allocated")
+        if event.name == "memo" and "uri" in event.context:
+            if args.verbose:
+                self.log(jobid, ts, "waiting for instance")
+
+            #  Wait for child instance to start rc2 using state-machine.wait
+            #  Then stop the reactor to return to caller.
```

Comment:
It would be slightly more accurate to say "Wait for child instance to complete rc1...", since after the RPC succeeds we just know that the RUN state was entered (rc2 would be started  in that state, if configured, but we don't know if it's happened yet).

Note that the RPC fails if the broker is in a state that cannot lead to RUN (e.g. if rc1 fails, it skips to CLEANUP)

Reply:
Thanks! Note that when this RPC fails an exception should be raised from Python (i.e. the "return value" of the RPC is implicitly checked).  I should add a test for this in the testsuite (right now the only test ensures that the command fails when the broker is given an invalid argument)

Reply:
Just verifying this works: the RPC `.get()` throws an exception on failure, but the message would probably confuse a normal user:
```
FileNotFoundError: [Errno 2] broker has surpassed RUN state
```
This error should probably be caught and rewritten to something more clear.